bundles/org.eclipse.swt/Eclipse SWT Custom Widgets/common/org/eclipse/swt/custom/CTabFolderRenderer.java

```diff
@@ -355,9 +357,15 @@ protected Point computeSize (int part, int state, GC gc, int wHint, int hHint) {
 							gc.setFont(gcFont);
 						}
 					}
+
+					width += getTextPadding(item, state) * 2;
+
 					if (parent.showClose || item.showClose) {
 						if ((state & SWT.SELECTED) != 0 || parent.showUnselectedClose) {
-							if (width > 0) width += INTERNAL_SPACING;
+							if (((state & SWT.SELECTED) != 0 && parent.showSelectedImage)
+									|| ((state & SWT.SELECTED) == 0 && parent.showUnselectedImage)) {
+								if (width > 0) width += INTERNAL_SPACING;
+							}
```

Comment:
This is rather nitpicky, but it feels to me like there needs to be less spacing here when no images are shown:
```suggestion
							} else {
								if (width > 0) width -= INTERNAL_SPACING;
							}
```
Actually, it is not about `INTERNAL_SPACING` to be subtracted but about the margin of the close button (which is equal to `INTERNAL_SPACING = 4`). This feels more natural to me, as I would expect the distance between left end of the tab and the beginning of the text should be equal to the distance between the end of the text and the beginning of the icon (and not until the beginning of the icon's margin).

Reply:
Incorporated the suggestion and following is how it looks like after the change:
<img width="500" alt="image" src="https://github.com/eclipse-platform/eclipse.platform.swt/assets/57699330/9f7e3c82-a3ab-4553-ac15-5d4cf5cf0bed">